### DISCUSSION TITLE
Docker / K8s Capabilities

### DISCUSSION THREAD

**Discussion by grairdon88** [Mimetis/Dotmim.Sync#717] [Q&A] [answered]
Hello,

I'm doing some research on this framework and have some questions around containerization and orchestration with Docker and Kubernetes.

1.  I do not see any documentation on using this in a containerized application, has anyone utilized this framework before in a application running in a Docker container?
2.  If there have been implementations of the framework in containerized applications, any issues came across while hosting in a cloud-hosted Kubernetes cluster?
3. Does this support .NET 6?

I appreciate any feedback anyone can provide. 

Thanks,
Greg Rairdon

**Comment by Mimetis** [comment, marked as answer]
I did not test **DMS** with **Kubernetes** yet.
Basically, the server part is a "_simple_" **Asp.Net Core Web API**, with a single **HTTP** endpoint.
So far, no reason it won't work.

The only concern is that the Web Api is not stateless, we need a session to be able to sync a whole database in chunks
If you want to test it in a K8s environment be sure, you have a reverse proxy with a session affinity.

**DMS** needs a local folder to store tmp files, should not be a problem, I guess (and the tmp folder is used only during a session sync, then deleted)

If you made some tests with K8s, let me know, I worked a lot with kubernetes in the last few years, and I'm interested in the results

Yes, **DMS** supports **.Net 6**

**Comment by grairdon88** [comment]
@Mimetis thank you for your timely reply!

My initial hunch is what you pointed out here: it's a .net core web api, it should be able to work. However to your point, because of local storage for tmp files, that'll be something that I'll need to research for container file storage.

I'll be happy to report back on K8s capabilities if as I evaluate this tool. Once again, I appreciate your feedback!
